src/components/audio-player-controls/audio-player-controls.tsx

```diff
@@ -10,91 +10,112 @@ const {Rewind, Forward, PlaylistButton, PlayPause, Volume, SpeedMenu} = ui.Compo
 const {
   redux: {useSelector}
 } = ui;
+const {Event} = ui;
 
 interface AudioPlayerControlsProps {
   pluginConfig: AudioPlayerConfig;
   player: any;
   onPluginsControlClick: () => void;
   showMorePluginsIcon: boolean;
+  eventManager: any;
 }
 
-const AudioPlayerControls = ({pluginConfig, player, onPluginsControlClick, showMorePluginsIcon}: AudioPlayerControlsProps) => {
+export const AudioPlayerControls = Event.withEventManager(({pluginConfig, player, onPluginsControlClick, showMorePluginsIcon, eventManager}: AudioPlayerControlsProps) => {
   const playlist = useSelector((state: any) => state.engine.playlist);
   const ref = useRef<HTMLDivElement>();
+  const [currentRate, setCurrentRate] = useState(player.playbackRate);
 
-  useEffect(() => {
-    ref.current?.setAttribute('tabindex', '0');
-  }, []);
+    useEffect(() => {
+      const element = ref.current;
+      if (!element || !eventManager) return;
+      element.setAttribute('tabindex', '0');
```

Comment:
instead of manipulating the ref, i would prefer to either wrap the speed menu component and add a listener on the ref, or to add an option to pass a listener into the speed menu component.

if the speed menu component is under our responsibility, we can update it instead of working with ref, which is intended more for runtime changes.